### PR TITLE
Feature: JS Api Docs

### PR DESCRIPTION
**Pull request by timotheeguerin**
This plugins typedoc with the markdown theme to extract the typescript api for a library.

### CODE REVIEW

File: docs/standard-library/rest/reference/js-api/interfaces/ResourceOperation.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[Documentation](../index.md) / ResourceOperation
```
Comment (timotheeguerin) [inline docs/standard-library/rest/reference/js-api/interfaces/ResourceOperation.md:1]: This is a lot of new files, do we want:
1. To actually include this in the committed repo? If not will have to make an extra step for generating the latest version vs next
2. This is using the markdown theme for typedoc, but we could get a more compact output like we have for the cadl if we wrote our own markdown theme

Comment (bterlson) [reply docs/standard-library/rest/reference/js-api/interfaces/ResourceOperation.md:1]: I would say no to having this checked in (feels like build output to me).

Comment (timotheeguerin) [reply docs/standard-library/rest/reference/js-api/interfaces/ResourceOperation.md:1]: Should we then also not have the cadl ref docs be included

Comment (allenjzhang) [reply docs/standard-library/rest/reference/js-api/interfaces/ResourceOperation.md:1]: That would be good. Let's use git ignore for all build output ref folders. For docs in the versioned release we do a manual step check-in or automate it with a small script easily.

**Review by allenjzhang — APPROVED** [review]

**Pull request merged**